extensions/wikia/Email/SpecialSendEmailController.class.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Email;
+
+
+class SpecialSendEmailController extends \WikiaSpecialPageController {
+
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+	const REQUIRED_PERMISSION = "staff";
+
+	public function __construct() {
+		parent::__construct( 'SendEmail', '', $listed = false );
+	}
+
+	/**
+	 * Run before index. First make sure the user can access page, then set the title and
+	 * add the css.
+	 */
+	public function init() {
+		$this->assertCanAccess();
+		$this->setTitle();
+		$this->addCss();
+	}
+
+	/**
+	 * Makes sure the current user is staff and can therefore access this page. If not, display
+	 * a permission error page.
+	 * @throws \PermissionsError
+	 */
+	private function assertCanAccess() {
+		if ( !$this->wg->User->isStaff() ) {
+			throw new \PermissionsError( self::REQUIRED_PERMISSION );
```

Comment:
Ah, good to know! Obviously I didn't have a full grasp of how permissions worked in Mediawiki, thanks for clarifying. I think I'll just go ahead and create a new right, since sendemail is already being used (and to test out my new Mediawiki access control knowledge :) ).